homeassistant/components/recorder/util.py

```diff
@@ -332,4 +333,4 @@ def perodic_db_cleanups(instance: Recorder):
     if instance.engine.dialect.name == "sqlite":
         # Execute sqlite to create a wal checkpoint and free up disk space
         _LOGGER.debug("WAL checkpoint")
-        instance.engine.execute("PRAGMA wal_checkpoint(TRUNCATE);")
+        instance.engine.connect().execute(text("PRAGMA wal_checkpoint(TRUNCATE);"))
```

Comment:
https://docs.sqlalchemy.org/en/14/core/connections.html#sqlalchemy.engine.Engine.connect

Reply:
Would it be cleaner to use a context to return the connection to the pool?

```python
with instance.engine.connect() as connection:
  connection.execute(text("PRAGMA wal_checkpoint(TRUNCATE);"))
```

Reply:
100%!